django/core/management/commands/makemigrations.py

```diff
@@ -295,10 +295,17 @@ def all_items_equal(seq):
                 subclass = type("Migration", (Migration,), {
                     "dependencies": [(app_label, migration.name) for migration in merge_migrations],
                 })
-                migration_name = "%04i_%s" % (
-                    biggest_number + 1,
-                    self.migration_name or ("merge_%s" % get_migration_name_timestamp())
-                )
+                parts = ['%04i' % (biggest_number + 1)]
+                if self.migration_name:
+                    parts.append(self.migration_name)
+                else:
+                    parts.append('merge')
+                    leaf_names = '_'.join(sorted(migration.name for migration in merge_migrations))
+                    if len(leaf_names) > 100:
```

Comment:
I'm happy to tweak this threshold value higher or lower if that is preferred.

Reply:
[Adam proposed 42](https://github.com/django/django/pull/12799#issuecomment-636778885). :thinking:

Reply:
Sounds good. I changed with 42 but can adjust to match Adam's work. If this becomes a common value, I can move it to a constant too.